owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/PublicShareDialogFragment.kt

```diff
@@ -610,11 +610,9 @@ class PublicShareDialogFragment : DialogFragment() {
 
         override fun onCancelDatePicker() {
 
-            val expirationToggle = view!!.findViewById<SwitchCompat>(R.id.shareViaLinkExpirationSwitch)
-
             // If the date has not been set yet, uncheck the toggle
-            if (expirationToggle.isChecked && shareViaLinkExpirationValue?.text === "") {
-                expirationToggle.isChecked = false
+            if (shareViaLinkExpirationSwitch.isChecked && shareViaLinkExpirationSwitch?.text === "") {
```

Comment:
`shareViaLinkExpirationSwitch?.text.isEmpty()` is not enough here?

Reply:
BTW i detected a problem:
Second check should not be `shareViaLinkExpirationSwitch?.text` but `shareViaLinkExpirationValue?.text`